File: FRONTEND CLIENT/HMS_CaseStudy_FrontEnd/src/services/BookingService.js
// BookingService.js
import axios from 'axios';

const BASE_URL = 'http://localhost:8082/api/bookings';

class BookingService {
  getAllBookings() {
    return axios.get(BASE_URL);
  }

  getBookingById(id) {
    return axios.get(`${BASE_URL}/${id}`);
  }

  addBooking(data) {
    return axios.post(BASE_URL, data);
  }

  updateBooking(id, data) {
    return axios.put(`${BASE_URL}/${id}`, data);
  }

  deleteBooking(id) {
    return axios.delete(`${BASE_URL}/${id}`);
  }
}

export default new BookingService();

File: BACKEND SERVER/BackEnd/Booking/src/main/java/com/microservice/booking/service/BookingService.java
package com.microservice.booking.service;


import java.util.List;

import com.microservice.booking.entity.Booking;

public interface BookingService {

    List<Booking> getAllBookings();

    Booking getBookingById(String id);

    Booking addBooking(Booking booking);

    Booking updateBooking(String id, Booking booking);

    void deleteBooking(String id);
}
